pinot-core/src/main/java/org/apache/pinot/core/data/manager/TableDataManager.java

```diff
@@ -79,6 +79,16 @@ void addSegment(@Nonnull String segmentName, @Nonnull TableConfig tableConfig,
    */
   void removeSegment(@Nonnull String segmentName);
 
+  /**
+   * Track a deleted segment.
+   */
+  void deleteSegment(@Nonnull String segmentName);
```

Comment:
Can we rename this function? `deleteSegment` sounds like we are deleting segments instead of book-keeping.  It's confusing since we have another method called `removeSegment`.

Reply:
+1

Reply:
Done. Renamed to trackDeletedSegment